core/jvmTest/src/kotlinx/serialization/CachingTest.kt

```diff
@@ -43,4 +43,21 @@ class CachingTest {
 
         assertEquals(1, factoryCalled)
     }
+
+    @Serializable
+    class Target
```

Comment:
There are no other branches of intrinsic, like value classes, e.t.c?

It may not be enough to check only the intrinsic of a regular class `class Target`

Reply:
There aren't any branches depending on class type. Either intrinsics are triggered and replacing `serializer<T>()` with _something_, or they are not. We have different branches for caches though (parametrized vs non-parametrized), but that is not what being tested here.